Sources/protoc-gen-swift/Descriptor+Extensions.swift

```diff
@@ -298,4 +299,60 @@ extension EnumDescriptor {
     }
     return nil
   }
+
+  /// Helper object that computes the alias relationships of
+  /// `EnumValueDescriptor`s for a given `EnumDescriptor`.
+  final class ValueAliasInfo {
+    /// The `EnumValueDescriptor`s that are not aliases of another value. In
+    /// the same order as the values on the `EnumDescriptor`.
+    let mainValues : [EnumValueDescriptor]
+
+    /// Find the alias values for the given value.
+    ///
+    /// - Parameter value: The value descriptor to look up.
+    /// - Returns The list of value descriptors that are aliases for this
+    ///     value, or `nil` if there are no alias (or if this was an alias).
+    func aliases(_ value: EnumValueDescriptor) -> [EnumValueDescriptor]? {
+      assert(mainValues.first!.enumType === value.enumType)
+      return aliasesMap[value.index]
+    }
+
+    /// Find the original for an alias.
+    ///
+    /// - Parameter value: The value descriptor to look up.
+    /// - Returns The original/main value if this was an alias otherwise `nil`.
+    func original(of: EnumValueDescriptor) -> EnumValueDescriptor? {
```

Comment:
I couldn't come up name for this one, open to suggestions.

Reply:
This seems fine; `originalValue(of:)` would send the wrong signal (it's returning the descriptor, not the value) and `originalValueDescriptor(of:)` is too verbose IMO.